Gems/Multiplayer/Code/Source/NetworkEntity/NetworkEntityManager.cpp

```diff
@@ -467,6 +467,54 @@ namespace Multiplayer
         return netEntityId;
     }
 
+    AzFramework::EntitySpawnTicket NetworkEntityManager::RequestNetSpawnableInstantiation(const AZ::Data::Asset<AzFramework::Spawnable>& rootSpawnable)
+    {
+        const AzFramework::Spawnable::EntityList& entityList = rootSpawnable->GetEntities();
+        if (entityList.size() == 0)
+        {
+            AZ_Assert(false, "RequestNetSpawnableInstantiation: No entities in the spawnable %s", rootSpawnable.GetHint().c_str())
+            return {};
+        }
+
+        // The first entity in every spawnable is the root one 
+        const AZ::Entity* rootEntity = (entityList.begin())->get();
+
+        const auto* holderComponent = rootEntity->FindComponent<NetworkSpawnableHolderComponent>();
+        if(!holderComponent)
+        {
+            // This spawnable doesn't have a corresponding network spawnable.
+            return {};
+        }
+
+        // Retrieve the corresponding network spawnable asset
+        AZ::Data::Asset<AzFramework::Spawnable> netSpawnableAsset = holderComponent->GetNetworkSpawnableAsset();
+
+        // Prepare the parameters for the spawning process
+        AzFramework::SpawnAllEntitiesOptionalArgs optionalArgs;
+        optionalArgs.m_priority = AzFramework::SpawnablePriority_High;
+
+        // Pre-insertion callback allows us to do network-specific setup for the entities before they are added to the scene
+        optionalArgs.m_preInsertionCallback = [spawnableAssetId = netSpawnableAsset.GetId()](AzFramework::EntitySpawnTicket::Id,
+            AzFramework::SpawnableEntityContainerView entities)
+        {
+            for (uint32_t netEntityIndex = 0; netEntityIndex < entities.size(); netEntityIndex++)
+            {
+                AZ::Entity* netEntity = *(entities.begin() + netEntityIndex);
+                AZ::Name spawnableName = AZ::Interface<INetworkSpawnableLibrary>::Get()->GetSpawnableNameFromAssetId(spawnableAssetId);
```

Comment:
Getting the `spawnableName` from the `spawnableAssetId` could be moved outside the loop, and maybe even outside the lambda function since it seems the `spawnableName` it's all the lambda needs.

Reply:
Yeah, good point! updated now, thanks!